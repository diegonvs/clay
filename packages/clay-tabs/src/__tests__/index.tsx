/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import '@testing-library/jest-dom/extend-expect';
import ClayTabs from '..';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';

const ClayTabsWithItems = () => {
	const [activeTabKeyValue, setActiveTabKeyValue] = React.useState<number>(0);

	const onClick = (val: number) => setActiveTabKeyValue(val);

	return (
		<>
			<ClayTabs>
				<ClayTabs.Item
					active={activeTabKeyValue == 0}
					onClick={onClick}
					tabName="Dummy1"
					tabkey={0}
				/>
				<ClayTabs.Item
					active={activeTabKeyValue == 1}
					data-testid="tabItem2"
					onClick={onClick}
					tabName="Dummy2"
					tabkey={1}
				/>
				<ClayTabs.Item
					active={activeTabKeyValue == 2}
					onClick={onClick}
					tabName="Dummy3"
					tabkey={2}
				/>
			</ClayTabs>
			<ClayTabs.Content activeTabKey={activeTabKeyValue}>
				<ClayTabs.TabPane data-testid="tabPane1" tabkey={0}>
					{'Tab Content 1'}
				</ClayTabs.TabPane>
				<ClayTabs.TabPane data-testid="tabPane2" tabkey={1}>
					{'Tab Content 2'}
				</ClayTabs.TabPane>
				<ClayTabs.TabPane tabkey={2}>
					{'Tab Content 3'}
				</ClayTabs.TabPane>
			</ClayTabs.Content>
		</>
	);
};

describe('ClayTabs', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(<ClayTabs />);

		expect(container).toMatchSnapshot();
	});

	it('renders justifying nav items', () => {
		const {container} = render(<ClayTabs justified />);

		expect(container).toMatchSnapshot();
	});

	it('renders with modern style', () => {
		const {container} = render(<ClayTabs modern />);

		expect(container).toMatchSnapshot();
	});

	it('renders with items', () => {
		const {container} = render(<ClayTabsWithItems />);

		expect(container).toMatchSnapshot();
	});

	it('renders nav items as anchors', () => {
		const {container, getByTestId} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item
						active
						component="a"
						tabName="One"
						tabkey={1}
					/>
					<ClayTabs.Item component="a" tabName="Two" tabkey={2} />
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane tabkey={1}>
						{'Content One'}
					</ClayTabs.TabPane>
					<ClayTabs.TabPane tabkey={2}>
						{'Content Two'}
					</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		const tabItem = getByTestId('oneTabItem');

		expect(tabItem.nodeName).toBe('A');

		expect(container).toMatchSnapshot();
	});

	it('renders disabled nav items', () => {
		const {container} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item active tabName="One" tabkey={1} />
					<ClayTabs.Item tabName="Two" tabkey={2} />
					<ClayTabs.Item disabled tabName="Three" tabkey={3} />
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane tabkey={1}>
						{'Content One'}
					</ClayTabs.TabPane>
					<ClayTabs.TabPane tabkey={2}>
						{'Content Two'}
					</ClayTabs.TabPane>
					<ClayTabs.TabPane tabkey={3}>
						{'Content Three'}
					</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		expect(container).toMatchSnapshot();
	});

	it('emits a number when clicking an item', () => {
		const onClick = jest.fn();

		const {container, getByTestId} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item
						active
						onClick={onClick}
						tabName="One"
						tabkey={1}
					/>
					<ClayTabs.Item
						component="a"
						onClick={onClick}
						tabName="Two"
						tabkey={2}
					/>
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane tabkey={1}>
						{'Content One'}
					</ClayTabs.TabPane>
					<ClayTabs.TabPane tabkey={2}>
						{'Content Two'}
					</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		fireEvent.click(getByTestId('oneTabItem'));
		expect(onClick).toBeCalled();
		expect(onClick).toBeCalledWith(1);

		fireEvent.click(getByTestId('twoTabItem'));
		expect(onClick).toBeCalled();
		expect(onClick).toBeCalledWith(2);
	});
});
