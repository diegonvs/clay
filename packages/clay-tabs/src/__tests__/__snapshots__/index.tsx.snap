// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayTabs renders 1`] = `
<div>
  <ul
    class="nav nav-tabs"
    role="tablist"
  />
</div>
`;

exports[`ClayTabs renders disabled nav items 1`] = `
<div>
  <ul
    class="nav nav-tabs"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <button
        aria-controls="one"
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active btn btn-unstyled"
        data-testid="oneTabItem"
        role="tab"
        type="button"
      >
        One
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-controls="two"
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="twoTabItem"
        role="tab"
        type="button"
      >
        Two
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-controls="three"
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled disabled"
        data-testid="threeTabItem"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Three
      </button>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane active"
      role="tabpanel"
      tabkey="1"
    >
      Content One
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
      tabkey="2"
    >
      Content Two
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
      tabkey="3"
    >
      Content Three
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders justifying nav items 1`] = `
<div>
  <ul
    class="nav nav-justified nav-tabs"
    role="tablist"
  />
</div>
`;

exports[`ClayTabs renders nav items as anchors 1`] = `
<div>
  <ul
    class="nav nav-tabs"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <a
        aria-controls="one"
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active"
        data-testid="oneTabItem"
        role="tab"
      >
        One
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        aria-controls="two"
        aria-disabled="true"
        aria-selected="false"
        class="nav-link"
        data-testid="twoTabItem"
        role="tab"
      >
        Two
      </a>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane active"
      role="tabpanel"
      tabkey="1"
    >
      Content One
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
      tabkey="2"
    >
      Content Two
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders with items 1`] = `
<div>
  <ul
    class="nav nav-tabs"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <button
        aria-controls="dummy1"
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active btn btn-unstyled"
        data-testid="dummy1TabItem"
        role="tab"
        type="button"
      >
        Dummy1
      </button>
    </li>
    <li
      class="nav-item"
      data-testid="tabItem2"
    >
      <button
        aria-controls="dummy2"
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="dummy2TabItem"
        role="tab"
        type="button"
      >
        Dummy2
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-controls="dummy3"
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="dummy3TabItem"
        role="tab"
        type="button"
      >
        Dummy3
      </button>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane active"
      data-testid="tabPane1"
      role="tabpanel"
      tabkey="0"
    >
      Tab Content 1
    </div>
    <div
      class="tab-pane"
      data-testid="tabPane2"
      role="tabpanel"
      tabkey="1"
    >
      Tab Content 2
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
      tabkey="2"
    >
      Tab Content 3
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders with modern style 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  />
</div>
`;
